content/blog/deploying-a-hugging-face-llm-in-hpe-private-cloud-ai.md

```diff
@@ -0,0 +1,135 @@
+
+Next up let's take the now deployed model that's ready for inference and connect to it and interact with it from a Jupyter Notebook.
+
+### Building the Jupyter Notebook
+
+First install `openai` if you do not already have it and import.
+
+```python
+# vLLM Chat OpenAI
+# !pip intall openai
+from openai import OpenAI
+```
+
+Then using the endpoint and key genered by HPE MLIS enter them into your Jupyter Notebook. Be sure to append /v1 to the URL.
+
+```python
+# Grab endpoint URL and API key from MLIS, remember to include "/v1" for latest version of the OpenAI-compatible API
+model = "HuggingFaceTB/SmolLM2-1.7B-Instruct"
+openai_api_base = "https://smollm2-1-7b-vllm-predictor-dave-wright-hpe-1073f7cd.hpepcai-ingress.pcai.hpecic.net/v1"
+openai_api_key = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE3Mzk5MzgzMzAsImlzcyI6ImFpb2xpQGhwZS5jb20iLCJzdWIiOiI5MjNhM2JhOC1mMGU4LTQxOTQtODNkMS05ZWY4NzNjZGYxOWYiLCJ1c2VyIjoiZGF2ZS53cmlnaHQtaHBlLmNvbSJ9.YwH9gGPxTWxy4RSdjnQA9-U3_u7P0OIcarqw25DV8bOiftU1L4IvvyERHspj2lMGtZWbff1F3uh84wjAePHaHDcDTLoGtq6gJYwo_qRU03xV8Q2lwBetCCLUE4OHqS608gjJ-j1SLyqwxFxlXkqMOtnBY5_nswlAwCzHV28P8u8XxxfWuXFmoJpSA1egCWVVfEoTuK8CTz9kUJJ5opSp6m8qdqJmC2qxH0igcpKmL2H_MZ-62UHfEf240VRtc0DRNlOjeCoDM79aVPs3SjCtGeVkeEHimJwJbfGFIcu3LibX3QjbABUzWb5BPPZjzyEYUVM5ak12_sJ8j1mUW-r0sA"
+```
+
+Now we need an OpenAI client interface.
+
+```python
+# create OpenAI client interface
+client = OpenAI(
+    api_key=openai_api_key,
+    base_url=openai_api_base,
+)
+```
+
+In order to interact with the model we need to create a chat function, but let's give it a history feature as well as basic chat.
+
+```python
+# Interactive chat function with message history. 
+def chat():
+    # Initialize conversation history
+    messages = []
+    
+    print("Chat with "+model+"! Type 'quit' to exit.")
+    
+    while True:
+        # Get user input
+        user_input = input("\nYou: ").strip()
+        
+        # Check for quit command
+        if user_input.lower() == 'quit':
+            print("Goodbye!")
+            break
+        
+        # Add user message to history
+        messages.append({"role": "user", "content": user_input})
+        
+        try:
+            # Get model response using chat completion
+            response = client.chat.completions.create(
+                model=model,
+                messages=messages
+            )
+            
+            # Extract assistant's message
+            assistant_message = response.choices[0].message.content
+            
+            # Add assistant's response to history
+            messages.append({"role": "assistant", "content": assistant_message})
+            
+            # Print the response
+            print("\nAssistant:", assistant_message)
+            
+        except Exception as e:
+            print(f"\nError: {str(e)}")
+```
+
+![Jupyter Notebook showing imported model endpoint and API key.](/img/jupyter.png)
+
+Then interacting with the model is done through a simple chat.
```

Comment:
Recommend rephrasing to "Once this is done, you can interact with the model through a simple chat."